ui/src/services/k8s/solutions.js

```diff
@@ -119,7 +119,17 @@ export async function getEnvironmentConfigMap(environment) {
 }
 
 export async function addSolutionToEnvironment(namespace, solName, solVersion) {
-  const patch = [{ op: 'add', path: `/data/${solName}`, value: solVersion }];
+  const result = await coreV1.readNamespacedConfigMap(
+    ENVIRONMENT_CONFIGMAP_NAME,
+    namespace,
+  );
+  let patch;
+  // the first patch we need to create data key. otherwise we will get 422 patch error.
```

Comment:
Or simply: `// Create the `data` key if it doesn't exists, otherwise we can't insert the solution field inside`